Templates/DefaultProject/Template/ExportScripts/export_standalone_monolithic.py

```diff
@@ -0,0 +1,321 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+import argparse
+import glob
+import logging
+import os
+import o3de.export_project as exp
+import pathlib
+import platform
+import shutil
+import sys
+import time
+import psutil
+
+from o3de.export_project import process_command
+from o3de import manifest
+from typing import List
+
+
+# This script is meant to be run via o3de cli export-project command, not from Python directly!
+# This export script operates in a project-centric manner, meaning all build artifacts are centered around the project folder. 
+# In order to use this script, the command should look like the following (on windows use o3de.bat, on linux use o3de.sh):
+
+# cd path\to\o3de\engine
+# .\scripts\o3de.bat  export-project --export-scripts path\to\o3de\ExportScripts\export_standalone_monolithic_project_centric.py \
+# --project-path path\to\project --log-level INFO --output-path path\to\output --build-non-mono-tools --config release --archive-output zip \
+# --seedlist path\to\seedlist1.seed --seedlist path\to\seedlist2.seed <etc..> 
+
+# The following parameters are optional, but may be helpful:
+#   seedlist: an O3DE seedlist for describing what assets of the project should be bundled
+#   gem-folder-name-to-copy: A Gem Folder from <ProjectRoot>/Gems that should be copied to the output location of the exported project
+#                               Files found in these folders will go into <output-path>/Gems/GemFolderName
+#                               Variants for game/server launcher only are game-gem-folder-name-to-copy/server-gem-folder-name-to-copy resp.
+#   project-file-pattern-to-copy: Any regex pattern starting from <ProjectRoot> that indicates what files should be copied to output location.
+#                                   Files found in these patterns will go into <output-path>
+#                                   Variants for game/server launcher only are game-file-pattern-to-copy/server-file-pattern-to-copy resp.
+
+# For more information on the available parameters for this script, check the parse_arguments function defined at the bottom of the file
+
+def  build_non_monolithic_code_modules( project_path: pathlib.Path,
+                                        non_mono_build_path: pathlib.Path):
+    #build the pre_requisite tools if needed
+    os.makedirs(non_mono_build_path, exist_ok=True)
+
+    process_command(["cmake", "-B", non_mono_build_path, "-S", project_path])
+
+    process_command(["cmake", "--build", non_mono_build_path, "--target", "AssetProcessorBatch", "AssetBundlerBatch", "--config", "profile", "--", "/m"])
```

Comment:
"/m" is visual studio specific. In fact, everything after `--` are arguments that are passed through to the underlying compiler. Instead of auto-building it, i think  and error should be raised to inform the user to build `AssetProcessorBatch` manually.

Reply:
I dont think requiring manual build of AssetProcessorBatch is appropriate for this script, since the purpose is to automate the entire build process, including dependencies. That being said, there is a `build-non-mono-tools` flag, so I could rework it so that if it's enabled and it does not detect the executable, it can rebuild it. Otherwise if it isnt there, it could error out.

I'll remove the visual studio specific parts.

Reply:
If we are going to perform long steps like building in this script, there should be at least a status messaging to stdout informing that the process is underway. Also, does the stdout/stderr streams captured or outputted to the console? or does it get eaten up by this script?

Reply:
`process_command` does indicate a status that the process is underway, if the user sets the log level to `INFO` in the arguments. And yes, stdout/stderr is outputted to the console.

Reply:
So to clarify, the function on it's own should not auto-fail and require manual building. The purpose of the function is to build the tools if desired.

In the main project centric export function, we have the following code snippet:
```python

    if should_build_all_code:
        if should_build_non_mono_tools:
            build_non_monolithic_code_modules(project_path, non_mono_build_path)
        
        build_monolithic_code_modules(project_path, project_name, build_config, mono_build_path,
                                      should_build_game_launcher, should_build_server_launcher, should_build_unified_launcher,
                                      allow_registry_overrides)

```
So if the user desired to build non_mono_tools, then building asset processor and batch are expected. But in both cases, I can add a check for the existence of the tool binaries before proceeding on. If those two are not present at the expected path, then the process failed, and manual intervention is required.